packages/tx/src/baseTransaction.ts

```diff
@@ -84,22 +85,22 @@ export abstract class BaseTransaction<TransactionObject> {
 
   constructor(txData: TxData | AccessListEIP2930TxData | FeeMarketEIP1559TxData) {
     const { nonce, gasLimit, to, value, data, v, r, s, type } = txData
-    this._type = new BN(toBuffer(type)).toNumber()
+    this._type = Number(bufferToBigInt(toBuffer(type)))
 
     const toB = toBuffer(to === '' ? '0x' : to)
     const vB = toBuffer(v === '' ? '0x' : v)
     const rB = toBuffer(r === '' ? '0x' : r)
     const sB = toBuffer(s === '' ? '0x' : s)
 
-    this.nonce = new BN(toBuffer(nonce === '' ? '0x' : nonce))
-    this.gasLimit = new BN(toBuffer(gasLimit === '' ? '0x' : gasLimit))
+    this.nonce = bufferToBigInt(toBuffer(nonce === '' ? '0x' : nonce))
```

Comment:
Also just to mention here: as @ryanio noted in some chat as well, on these kind of initializations the `0` - or sometimes empty - input values (also: in it's different type flavors) are really critical and we already stumbled upon that (`v` being initially empty on initialization ended up with `BN(0)` and therefore let `isSigned()` show `true` 😬).

So there should be an additional look if tests like `inputValue.spec.ts` really cover these cases and are adopted adequately. In doubt it also can't hurt to write 2-3 additional test cases.